internal/extender/resource.go

```diff
@@ -196,24 +196,27 @@ func (s *SparkSchedulerExtender) fitEarlierDrivers(
 	return true
 }
 
-func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v1.Pod, nodeNames []string) (string, string, error) {
+func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v1.Pod, nodeNames set) (string, string, error) {
```

Comment:
I think we should preserve the signature and inline the convertToSet method below

Reply:
makes sense, I actually just extracted this into its own method, can you take a look again?